pkg/wireguard/wireguard.go

```diff
@@ -71,17 +73,18 @@ func syncAddress(_ agent.State, iface string) error {
 }
 
 func createLinkUsingUserspaceImpl(iface string, wgUserspaceImplementationFallback string) error {
-
-	bashCommand := fmt.Sprintf("mkdir -p /dev/net && if [ ! -c /dev/net/tun ]; then\n    mknod /dev/net/tun c 10 200\nfi && %s %s", wgUserspaceImplementationFallback, iface)
-	cmd := exec.Command("bash", "-c", bashCommand)
-
-	err := cmd.Run()
-	if err != nil {
-		return err
+	if err := os.MkdirAll("/dev/net", 0o600); err != nil {
+		return fmt.Errorf("mkdir all: %w", err)
 	}
 
-	return nil
-
+	if _, err := os.Stat("/dev/net/tun"); os.IsNotExist(err) {
+		if err := unix.Mknod("/dev/net/tun", unix.S_IFIFO|0o600, int(unix.Mkdev(10, 100))); err != nil {
```

Comment:
Would appreciate :eyes: on this part @jodevsa - I think it's correct.

https://github.com/slackhq/nebula/blob/50b24c102ebc01aae21f0bcfc2dfff08f0611fcf/overlay/tun_linux.go#L90

https://github.com/tailscale/tailscale/blob/b5dbf155b1b0fbd5947160d8bca4085c6ff039a5/cmd/containerboot/main.go#L810

Reply:
LGTM; It would be useful to try it out. We have basic integration-tests that runs on each PR but it only tests creating the resources. It doesn't really test connecting to the VPN and making sure connectivity works as expected :( https://github.com/jodevsa/wireguard-operator/blob/main/internal/it/it_test.go#L39

Reply:
Also, the default behaviour is to use kernal implementation of WG. I guess the environment where the IT runs supports the kernal implementation of WG. we can use https://github.com/jodevsa/wireguard-operator/blob/main/cmd/agent/main.go#L26 
`--wg-use-userspace-implementation` to test this feature manually or write an IT for it. What are your thoughts on this?

Reply:
btw, you can easily manually test your changes by building a release from your branch. You can access the following workflow which would do all the building for you :). You only need to run this workflow https://github.com/jodevsa/wireguard-operator/actions/workflows/manual-dev-release-workflow.yaml

Reply:
<img width="408" alt="Screenshot 2024-05-09 at 13 01 49" src="https://github.com/jodevsa/wireguard-operator/assets/14154314/09403e72-15ed-4fde-ba7b-8e57c2ccfca6">

here is an example :)

Reply:
after the workflow completes you can access the release.yaml file that you can apply to a your kubernetes cluster

<img width="1212" alt="Screenshot 2024-05-09 at 13 06 58" src="https://github.com/jodevsa/wireguard-operator/assets/14154314/250a94e9-695b-4530-81ee-2c722a9a2b8c">

Reply:
example: https://github.com/jodevsa/wireguard-operator/actions/runs/9016924722

Reply:
Thanks - I think that seems reasonable. Will try and test something this evening maybe.

Reply:
@jodevsa I tested this and it seems to work the same as the kernel version, such I can successfully connect to my router (192.168.1.1) from 5G through a wireguard tunnel.

Reply:
thanks alot!